_posts/2021-01-08-2020-review.md

```diff
@@ -0,0 +1,254 @@
+* [Securing Trino with Dain]({% post_url 2020-08-13-training-security %})
+* [Configuring and Tuning Trino with Dain]({% post_url 2020-08-27-training-performance %})
+
+## Features and code
+
+While everything mentioned is already exciting, the true work is visible in the
+new features and improvements in Trino. It is a long list, but read on. You
+won’t want to miss anything.
+
+### Improvements to ANSI SQL support
+
+A core feature of Trino is the ability to use the same standard SQL for any
+connected data source. These improvements empower all users.
+
+* Variable-precision temporal types, with precision down to picoseconds. This a
+  very important feature for any time critical systems such as financial
+  transactions processing
+* Correct, and now SQL spec compliant Timestamp semantics, making migration of
+  SQL statements from other compliant systems such as many RDBMSs easier
+* Implicit coercions for `INSERT` clause.
+* Support for `RANGE` and `GROUPS`-based window frames
+* More support for various shapes of correlated subqueries
+* Support for `INTERSECT ALL` and `EXCEPT ALL`
+* Experimental support for [recursive queries](/docs/current/sql/select.html?highlight=recursive#with-recursive-clause)
+* Enforcement of `NOT NULL` constraints when inserting data
+* Quantified comparisons (e.g., > ALL (...)) in aggregation
+
+### Other query improvements
+
+A number of other features were added to make querying your data sources with
+Trino even more powerful:
+
+* [T-digest data type](/docs/current/language/types.html#t-digest) and functions
+  for approximate quantile computations
+* Support for setting and reading column comments
+* Numerous new functions including `concat_ws()`, `regexp_count()`,
+  `regexp_position()`, `contains_sequence()`, `murmur3()`,
+  `from_unixtime_nanos()`, `from_iso8601_timestamp_nanos()`,
+  `human_readable_seconds()`, `bitwise` operations, `luhn_check()`,
+  `approx_most_frequent()`, `translate()`, `starts_with()`
+
+## Performance
+
+Trino is already [ludicrously fast](/index.html). But then again, even faster is
+better, so we worked on that:
+
+* Improved pushdown of complex operations into connectors, including
+  [aggregation and topN pushdown](/docs/current/optimizer/pushdown.html). We’ve
+  extended several connectors with these new capabilities.
+* [Dynamic filtering and partition pruning]({% post_url
+  2020-06-14-dynamic-partition-pruning %}), which can improve performance of
+  highly selective joins manyfold.
+* Cost-based decisions for queries containing `IN <subquery>` in `WHERE` clause
+* Information_schema performance improvements, which benefit third-party BI
+  tools that need to inspect table metadata (e.g., DBeaver, Datagrip, etc).
+* [Faster queries on nested data in Parquet and ORC]({% post_url
+  2020-08-14-dereference-pushdown %})
+* Faster and more accurate `approx_percentile`, based on t-digest data structure
```

Comment:
what's the spelling? "t-Digest" ??

Reply:
I don't know. I've seen "t-digest", "t-Digest", "T-digest".

Reply:
Yeah .. same..